api/v1alpha1/nutanix_node_types.go

```diff
@@ -60,6 +60,24 @@ type NutanixMachineDetails struct {
 	// The project must already be present in the Prism Central.
 	// +kubebuilder:validation:Optional
 	Project *NutanixResourceIdentifier `json:"project,omitempty"`
+
+	// List of GPU devices that need to be added to the machines.
+	// +kubebuilder:validation:Optional
+	GPUs []NutanixGPU `json:"gpus,omitempty"`
+}
+
+type NutanixGPU struct {
```

Comment:
Whats the reason to have a new type here over the upstream `capxv1.NutanixGPU`? The other fields are already using the upstream types directly?

Reply:
using upstream types now